lib/commons/dom/is-in-text-block.js

```diff
@@ -84,24 +92,27 @@ function isInTextBlock(node) {
     ) {
       return false;
 
-      // Don't walk links, we're only interested in what's not in them.
-    } else if (
-      (nodeName === 'A' && currNode.href) ||
-      (currNode.getAttribute('role') || '').toLowerCase() === 'link'
-    ) {
-      if (currNode === node) {
-        inBrBlock = 1;
-      }
+      // Don't walk widgets, we're only interested in what's not in them.
+    } else if (isWidget(currNode)) {
       // Grab all the text from this element, but don't walk down it's children
-      linkText += currNode.textContent;
+      widgetText += currNode.textContent;
       return false;
     }
   });
 
   parentText = sanitize(parentText);
-  linkText = sanitize(linkText);
+  if (options?.noLengthCompare) {
+    return parentText.length !== 0;
+  }
+
+  widgetText = sanitize(widgetText);
+  return parentText.length > widgetText.length;
+}
 
-  return parentText.length > linkText.length;
+function isWidget(domNode) {
```

Comment:
This is the second instance of this function (the other being in `widget-not-inline-matches.js`, but they are slightly different. Thwidget-not-inline-matches.js` only accounts for `listbox` and `combobox` composite roles, but this one includes all composite roles. Should this function be pulled out and made a shared function that both use?

Reply:
Noticed that did you?... You're right. Need to fix that. I'll go find a better solution. I don't quite like how wonky this is with the listbox / combobox exception. I might just change what type we give to them and avoid needing a function at all.